tests/test_components_adq.py

```diff
@@ -174,3 +176,129 @@ def test_pull_baseline(mock_sqs_remi_run, in_dtype):
 
     for i, j in product(range(2), range(3)):
         np.testing.assert_allclose(out[i, j], single_trace - 24.5)
+
+
+def test_train_data(mock_sqs_remi_run):
+    ch = AdqRawChannel(mock_sqs_remi_run, '3B', digitizer='SQS_DIGITIZER_UTC2')
+    assert ch.raw_samples_key[0].ndarray()[0, 0] == -100
+
+    xr_data = ch.train_data(roi=(np.s_[:30000],))
+    assert isinstance(xr_data, xr.DataArray)
+    assert xr_data.dims == ('trainId', 'sample')
+
+    # Test unlabelled and non-tuple roi.
+    np_data = ch.train_data(labelled=False, roi=np.s_[:30000])
+    assert isinstance(np_data, np.ndarray)
+
+    for data in [xr_data.data, np_data]:
+        assert data.shape == (100, 30000)
+        assert data.dtype == np.float32
+
+        assert np.isclose(data[0, 0], 0.0)
+        assert np.isclose(data.min(), 0.0)
+        assert np.isclose(data.max(), 5.0)
+        assert np.all(data.argmax(axis=1) == 8974)
```

Comment:
The `numpy.testing` module has various helpers that can give more detailed information if the assertion fails - in particular `assert_array_equal` and `assert_allclose`. They allow both array/array comparisons and array/scalar.

```suggestion
        np.testing.assert_array_equal(data.argmax(axis=1), 8974)
```

Reply:
Hmmm, I'm regularly using that, not sure why I didn't here 🤔